			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-027

Bug-Reported-by:	Emanuel Haupt <ehaupt@FreeBSD.org>
Bug-Reference-ID:	<20221001123841.27e1bfb3bf3ed58ab32ea2b5@FreeBSD.org>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2022-10/msg00000.html

Bug-Description:

The configure test for the presence of strtoimax(3) is inverted.

Patch (apply with `patch -p0'):


*** ../bash-5.2-patched/m4/strtoimax.m4	Mon Apr 11 16:31:52 2022
--- m4/strtoimax.m4	Tue Apr  9 11:12:36 2024
***************
*** 30,34 ****
  ])
  AC_MSG_RESULT($bash_cv_func_strtoimax)
! if test $bash_cv_func_strtoimax = yes; then
  AC_LIBOBJ(strtoimax)
  fi
--- 30,34 ----
  ])
  AC_MSG_RESULT($bash_cv_func_strtoimax)
! if test $bash_cv_func_strtoimax = no; then
  AC_LIBOBJ(strtoimax)
  fi
*** ..//bash-5.2-patched/configure	Fri Sep 23 10:13:22 2022
--- configure	Tue Apr  9 11:13:21 2024
***************
*** 20444,20448 ****
  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $bash_cv_func_strtoimax" >&5
  printf "%s\n" "$bash_cv_func_strtoimax" >&6; }
! if test $bash_cv_func_strtoimax = yes; then
  case " $LIBOBJS " in
    *" strtoimax.$ac_objext "* ) ;;
--- 20444,20448 ----
  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $bash_cv_func_strtoimax" >&5
  printf "%s\n" "$bash_cv_func_strtoimax" >&6; }
! if test $bash_cv_func_strtoimax = no; then
  case " $LIBOBJS " in
    *" strtoimax.$ac_objext "* ) ;;

*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 26
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 27
  
  #endif /* _PATCHLEVEL_H_ */
